Math/NumberTheory/GaussianIntegers.hs

```diff
@@ -155,15 +158,20 @@ gcdG' g h
     | h == 0    = g --done recursing
     | otherwise = gcdG' h (abs (g `modG` h))
 
--- |Find a Gaussian integer whose norm is the given prime number.
+-- |Find a Gaussian integer whose norm is the given prime number using
+-- <http://www.ams.org/journals/mcom/1972-26-120/S0025-5718-1972-0314745-6/S0025-5718-1972-0314745-6.pdf Hermite-Serret algorithm>.
 findPrime' :: Integer -> GaussianInteger
-findPrime' p =
-    let (Just c) = Moduli.sqrtModMaybe (-1) (FieldCharacteristic (PrimeNat . integerToNatural $ p) 1)
-        k  = Powers.integerSquareRoot p
-        bs = [1 .. k]
-        asbs = map (\b' -> ((b' * c) `mod` p, b')) bs
-        (a, b) = head [ (a', b') | (a', b') <- asbs, a' <= k]
-    in a :+ b
+findPrime' p = case Moduli.sqrtModMaybe (-1) (FieldCharacteristic (PrimeNat . integerToNatural $ p) 1) of
```

Comment:
Not related to this PR, but it seems odd to have `findPrime'` when there is no `findPrime`. Is it worth renaming to `findPrime`, and keeping `findPrime'` around as a deprecated name?

Reply:
Good point. Will do.